ansible-scylla-node/tasks/upgrade/main.yml

```diff
@@ -150,26 +158,28 @@
 - name: Check if specified Scylla version can not be used as major upgrade
   ansible.builtin.fail:
     msg: "Version {{ scylla_upgrade['version'] }} specified for Scylla {{ scylla_upgrade['edition_friendly_name'] }} can't be used as a major upgrade since version detected is {{ scylla_detected['version'] }}."
+  any_errors_fatal: True
   when: >
-      upgrade_major and
+      scylla_version != 'latest' and upgrade_major and
```

Comment:
This comment is no surprise, but if you stop and think this creates a chicken and the egg situation somehow:

If the user specified that he wants a **major** upgrade, then it means that we must re-create the repositories. If we are to re-create the repositories, then it means that we are going to install something new, and at this point we have passed a validation step. It makes no sense to create a repository, to query whatever `latest` is just to fail later.

In fact, the original writer of the upgrade process very likely fell under the same thinking. When one specifies `latest` today, the value will be automatically be selected from whatever `scylla_support` first, this - in my view - is worse, because one may want to run a major upgrade that is not going to the latest and greatest major. :-)

So you need to decide: Whether we go through the whole pain of re-creating repos just for checking if latest is latest, or if we consider latest in general as a dangerous and non-recommended value. ;-)